### ISSUE TITLE
Add a Mid raw remap text field for delivery services.

### ISSUE THREAD

**Issue by traeak** [apache/trafficcontrol#5254]
## I'm submitting a ...

-  improvement request (usability, performance, tech debt, etc.)

## Traffic Control components affected ...

-  Documentation
-  Traffic Ops
-  Traffic Ops ORT
-  Traffic Portal
-  CI tests

## Current behavior:

Currently the Delivery Service page in TP has a "Raw Remap Text" field which just applies to the Edge remap rules.

## New behavior:

I'd like this to be split into:

Mid Raw Remap Text
and
Edge Raw Remap Text

This should allow experimental plugins and/or plugin options to be used without requiring a new release of TC.

**Comment by rawlinp** [comment]
With topologies, the traditional idea of a "mid" is changed. Would we want this raw remap to apply to basically all forward proxy caches? Do we need to differentiate between forward proxies that are _last_ in the hierarchy (just before the origin) and forward proxies that aren't (aka "inner")?

**Comment by rob05c** [comment]
Yeah, going forward this would have to be "inner" or "last" just like Header Rewrites.

But, I think we should think very hard about this and similar things, and see if there's a more structured way to do it.

Injecting raw text like this prone to typos, bugs, ops misunderstandings or not noticing it.

We should really see if there's a safer way to accomplish whatever the specific need here is.

**Comment by traeak** [comment]
There needs to be a way to configure ATS without requiring a new version of TC to do it.   This particular instance should allow new plugins and plugin options to be exercised.
